From f7c3199c6d52058b0a8e54b2a05d2efbc5fe88c3 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:53:25 -0400
Subject: [tools] perf/trace: Decode architecture-specific signal numbers

Message-id: <1408953210-26343-318-git-send-email-jolsa@redhat.com>
Patchwork-id: 88941
O-Subject: [PATCH RHEL7.1 BZ1133083 317/322] perf trace: Decode architecture-specific signal numbers
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 02c5bb4a352a4cca56e9b5d3a2a57d61062eb2e1
Author: Ben Hutchings <ben@decadent.org.uk>
Date: Thu Feb 6 01:00:41 2014 +0000

description
===========
SIGSTKFLT is not defined on alpha, mips or sparc.

SIGEMT and SIGSWI are defined on some architectures and should be
decoded here if so.

Fixes: 8bad5b0abfdb ('perf trace: Beautify signal number arg in several syscalls')
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index 6aa6fb6..f954c26 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -825,7 +825,6 @@ static size_t syscall_arg__scnprintf_signum(char *bf, size_t size, struct syscal
  P_SIGNUM(PIPE);
  P_SIGNUM(ALRM);
  P_SIGNUM(TERM);
- P_SIGNUM(STKFLT);
  P_SIGNUM(CHLD);
  P_SIGNUM(CONT);
  P_SIGNUM(STOP);
@@ -841,6 +840,15 @@ static size_t syscall_arg__scnprintf_signum(char *bf, size_t size, struct syscal
  P_SIGNUM(IO);
  P_SIGNUM(PWR);
  P_SIGNUM(SYS);
+#ifdef SIGEMT
+ P_SIGNUM(EMT);
+#endif
+#ifdef SIGSTKFLT
+ P_SIGNUM(STKFLT);
+#endif
+#ifdef SIGSWI
+ P_SIGNUM(SWI);
+#endif
  default: break;
  }
 
-- 
1.7.1